xls/modules/rle/rle_dec.x

```diff
@@ -0,0 +1,520 @@
+      (tok)
+    }(tok);
+    let RepeatTestOutputs: TestCommonSymbol[6] = [
+      TestCommonSymbol: 0xA, TestCommonSymbol: 0xA, TestCommonSymbol: 0xA,
+      TestCommonSymbol: 0xA, TestCommonSymbol: 0xA, TestCommonSymbol: 0xA,
+    ];
+    let tok = for ((counter, symbol), tok):
+        ((u32, TestCommonSymbol) , token)
+        in enumerate(RepeatTestOutputs) {
+      let last = counter == (array_size(RepeatTestOutputs) - u32:1);
+      let data_out = TestCommonDecOutData{symbol: symbol, last: last};
+      let (tok, dec_output) = recv(tok, dec_output_r);
+      trace_fmt!(
+          "Received {} transactions, symbol: 0x{:x}, last: {}",
+          counter + u32:1, dec_output.symbol, dec_output.last
+      );
+      assert_eq(dec_output, data_out);
+      (tok)
+    }(tok);
+    send(tok, terminator, true);
+    ()
+  }
+}
+
+// Check that RLE decoder will create 2 `last` output packets,
+// when 2 `last` input packets were consumed.
+#[test_proc]
+proc RunLengthDecoderLastAfterLastTest {
+  terminator: chan<bool> out;            // test termination request
+  dec_input_s: chan<TestCommonDecInData> out;
+  dec_output_r: chan<TestCommonDecOutData> in;
+
+  init {()}
+
+  config(terminator: chan<bool> out) {
+    let (dec_input_s, dec_input_r)   = chan<TestCommonDecInData>;
+    let (dec_output_s, dec_output_r) = chan<TestCommonDecOutData>;
+
+    spawn RunLengthDecoder<TEST_SYMBOL_WIDTH, TEST_COUNT_WIDTH>(
+        dec_input_r, dec_output_s);
+    (terminator, dec_input_s, dec_output_r)
+  }
+
+  next(tok: token, state: ()) {
+    let LastAfterLastTestStimuli: TestCommonDecInData[2] =[
+      TestCommonDecInData {
+        symbol: TestCommonSymbol:0x1,
+        count: TestCommonCount:0x1,
+        last:true
+      },
+      TestCommonDecInData {
+        symbol: TestCommonSymbol:0x2,
+        count: TestCommonCount:0x1,
+        last:true
+      },
+    ];
+    let tok = for ((counter, stimulus), tok):
+        ((u32, TestCommonDecInData) , token)
+        in enumerate(LastAfterLastTestStimuli) {
+      let tok = send(tok, dec_input_s, stimulus);
+      trace_fmt!("Sent {} stimuli, symbol: 0x{:x}, count:{}, last: {}",
+          counter + u32:1, stimulus.symbol, stimulus.count, stimulus.last);
+      (tok)
+    }(tok);
+    let LastAfterLastTestOutputs: TestCommonDecOutData[2] = [
+      TestCommonDecOutData{symbol: TestCommonSymbol: 0x1, last: true},
+      TestCommonDecOutData{symbol: TestCommonSymbol: 0x2, last: true},
+    ];
+    let tok = for ((counter, output), tok):
+        ((u32, TestCommonDecOutData) , token)
+        in enumerate(LastAfterLastTestOutputs) {
+      let (tok, dec_output) = recv(tok, dec_output_r);
+      trace_fmt!(
+          "Received {} transactions, symbol: 0x{:x}, last: {}",
+          counter + u32:1, dec_output.symbol, dec_output.last
+      );
+      assert_eq(dec_output, output);
+      (tok)
+    }(tok);
+    send(tok, terminator, true);
+    ()
+  }
+}
+
+// Check RLE decoder on longer transaction
```

Comment:
I think we can simply keep this test + the special cases (and drop the simpler one at the beginning), what do you think?

Reply:
Sure thing, I'll see that 2 tests are redundant. I'll remove them in next update

Reply:
I hope it better now, wdyt @proppy?

Reply:
yep, I don't think we need `RunLengthDecoderSingleSymbolTest` and `RunLengthDecoderSinglePacketTest`, since we have the longer test.

I think test that cover corner cases (`last`, `overflow` packet patterns) would be more relevant.

Reply:
I've removed these tests. I agree that they were redundant.